### PR TITLE
[SPARK-41963][CONNECT] Fix DataFrame.unpivot to raise the same error class when the `values` argument is empty

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Fixes `DataFrame.unpivot` to raise the same error class when the `values` argument is an empty list/tuple.

### Why are the changes needed?

Currently `DataFrame.unpivot` raises a different error class, `UNPIVOT_REQUIRES_VALUE_COLUMNS` for PySpark vs. `UNPIVOT_VALUE_DATA_TYPE_MISMATCH` for Spark Connect.

In `Unpivot`, an empty list/tuple as `values` argument is different from `None`. It should handle them differently.

### Does this PR introduce _any_ user-facing change?

`DataFrame.unpivot` on Spark Connect will raise the same error class as PySpark.

### How was this patch tested?

Enabled `DataFrameParityTests.test_unpivot_negative`.

### CODE REVIEW

File: connector/connect/common/src/main/protobuf/spark/connect/relations.proto [hunk 1]
```diff
@@ -729,13 +729,17 @@ message Unpivot {
   repeated Expression ids = 2;
 
   // (Optional) Value columns to unpivot.
-  repeated Expression values = 3;
+  optional Values values = 3;
```
Comment (HyukjinKwon) [inline connector/connect/common/src/main/protobuf/spark/connect/relations.proto:732]: cc @amaliujia @zhengruifeng FYI

**Review by HyukjinKwon — APPROVED** [review]
LGTM except that the protobuf definitoin part which I am not super used to.

**Review by zhengruifeng — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
oops, mind rebasing this? otherwise should be good to go.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.4.

**Pull request closed without merging**